sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OpenAiController.java

```diff
@@ -100,118 +66,84 @@ private static String objectToJson(@Nonnull final Object obj) {
   @GetMapping("/streamChatCompletion")
   @Nonnull
   ResponseEntity<ResponseBodyEmitter> streamChatCompletion() {
-    final var stream =
-        OpenAiClient.forModel(GPT_35_TURBO)
-            .withSystemPrompt("Be a good, honest AI and answer the following question:")
-            .streamChatCompletion(
-                "Can you give me the first 100 numbers of the Fibonacci sequence?");
-
-    final var emitter = new ResponseBodyEmitter();
-
-    final Runnable consumeStream =
-        () -> {
-          try (stream) {
-            stream.forEach(deltaMessage -> send(emitter, deltaMessage));
-          } finally {
-            emitter.complete();
-          }
-        };
-
-    ThreadContextExecutors.getExecutor().execute(consumeStream);
-
-    // TEXT_EVENT_STREAM allows the browser to display the content as it is streamed
-    return ResponseEntity.ok().contentType(MediaType.TEXT_EVENT_STREAM).body(emitter);
-  }
-
-  /**
-   * Send a chunk to the emitter
-   *
-   * @param emitter The emitter to send the chunk to
-   * @param chunk The chunk to send
-   */
-  public static void send(@Nonnull final ResponseBodyEmitter emitter, @Nonnull final String chunk) {
-    try {
-      emitter.send(chunk);
-    } catch (final IOException e) {
-      log.error(Arrays.toString(e.getStackTrace()));
-      emitter.completeWithError(e);
-    }
+    final var message = "Can you give me the first 100 numbers of the Fibonacci sequence?";
+    return service.streamChatCompletion(message);
   }
 
   /**
    * Chat request to OpenAI with an image
    *
-   * @return the assistant message response
+   * @return a ResponseEntity with the response content
    */
   @GetMapping("/chatCompletionImage")
   @Nonnull
-  OpenAiChatCompletionOutput chatCompletionImage() {
-    final var request =
-        new OpenAiChatCompletionParameters()
-            .addMessages(
-                new OpenAiChatUserMessage()
-                    .addText("Describe the following image.")
-                    .addImage(
-                        "https://upload.wikimedia.org/wikipedia/commons/thumb/5/59/SAP_2011_logo.svg/440px-SAP_2011_logo.svg.png",
-                        ImageDetailLevel.HIGH));
-
-    return OpenAiClient.forModel(GPT_4O).chatCompletion(request);
+  ResponseEntity<String> chatCompletionImage(
+      @RequestHeader(value = "accept", required = false) final String accept)
+      throws JsonProcessingException {
+    final var response =
+        service.chatCompletionImage(
+            "https://upload.wikimedia.org/wikipedia/commons/thumb/5/59/SAP_2011_logo.svg/440px-SAP_2011_logo.svg.png");
+    if (accept.equals("application/json")) {
```

Comment:
(Major?)

I think you should change all of the header checks in the following way, to avoid null pointer exceptions

```suggestion
    if ("application/json".equals(accept)) {
```